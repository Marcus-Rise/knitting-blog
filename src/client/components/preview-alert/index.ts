export { PreviewAlert } from "./preview-alert.component";
