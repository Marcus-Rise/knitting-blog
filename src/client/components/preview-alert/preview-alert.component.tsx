import type { FC } from "react";
import React from "react";
import Link from "next/link";
import styled from "styled-components";
import { colors } from "../../styles";

const Root = styled.div`
  position: sticky;
  top: 0;
  width: 100vw;
  padding: 0.05rem 2rem;
  background-color: ${colors.primary};
  color: white;
  font-size: 1.5rem;
  z-index: 2;

  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const LinkStyled = styled.a`
  text-decoration: none;
  font-size: 2rem;
  color: white;

  &:hover {
    cursor: pointer;
  }
`;

const PreviewAlert: FC<{ title: string }> = ({ title }) => (
  <Root>
    <h5>Режим предпросмотра записи: {`"${title}"`}</h5>
    <Link href={"/api/exit-preview"}>
      <LinkStyled title={"Выйти"}>X</LinkStyled>
    </Link>
  </Root>
);

export { PreviewAlert };
