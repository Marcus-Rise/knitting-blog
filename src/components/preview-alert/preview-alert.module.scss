.root {
  position: sticky;
  top: 0;
  width: 100vw;
  padding: .05rem 2rem;
  background-color: var(--primary);
  color: white;
  font-size: 1.5rem;
  z-index: 2;

  display: flex;
  justify-content: space-between;
  align-items: center;
}

.close {
  text-decoration: none;
  font-size: 2rem;
  color: white;

  &:hover {
    cursor: pointer;
  }
}
