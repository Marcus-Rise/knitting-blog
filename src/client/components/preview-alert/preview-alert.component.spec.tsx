import { render } from "@testing-library/react";
import React from "react";
import { PreviewAlert } from "./preview-alert.component";

jest.mock("next/link", () => ({
  __esModule: true,
  default: jest.fn((props) => <>{props.children}</>),
}));

describe("PreviewAlert", () => {
  test("render", () => {
    const { asFragment } = render(<PreviewAlert title={"Post title"} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
