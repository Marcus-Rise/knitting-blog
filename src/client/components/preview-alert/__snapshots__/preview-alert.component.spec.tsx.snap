// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewAlert render 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <h5>
      Режим предпросмотра записи: "Post title"
    </h5>
    <a
      class="close"
      title="Выйти"
    >
      X
    </a>
  </div>
</DocumentFragment>
`;
